client/src/app/shared/progress-bar/progress-bar.component.ts

```diff
@@ -0,0 +1,32 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { ThemePalette } from '@angular/material';
+import { filter, map } from 'rxjs/operators';
+import { ProgressBarEventType, ProgressBarService } from './progress-bar.service';
+
+/**
+ * As there is no export from material this type is added.
+ */
+type ProgressBarMode = 'determinate' | 'indeterminate' | 'buffer' | 'query';
```

Comment:
Curious: I would've prolly gone with an `enum`, but I'm also not very good at TypeScript. Does this custom type introduce any other characteristics I should be aware of? One thing I could imagine is that, if we'd assign a ProgressBar mode from the outside world, we don't **have** to use an enum, we can just stick to the string values.

Reply:
The only reason why I used a type is that it is used for type safety and doesn't create any generated code.

Reply:
Sounds reasonable!